src/main/scala/is/hail/expr/AST.scala

```diff
@@ -344,10 +366,10 @@ case class Select(posn: Position, lhs: AST, rhs: String) extends AST(posn, lhs)
         }
   }
 
-  def toIR(agg: Option[String] = None): Option[IR] = for {
+  def toIR(agg: Option[String] = None): ToIRErr[IR] = for {
     s <- lhs.toIR(agg)
-    t <- someIf(lhs.`type`.isInstanceOf[TStruct], lhs.`type`.asInstanceOf[TStruct])
-    f <- t.selfField(rhs)
+    t <- whenOfType[TStruct](lhs)
+    f <- fromOption(this, "$t must have field $rhs", t.selfField(rhs))
```

Comment:
I think this happens for things that were previously no argument, no parentheses methods like `object.noArgMethod`, but I'm not certain.

I think the python API was designed to prevent user-errors like `mt.Af` (instead of `mt.AF`) from getting to this level.